File: src/main/java/idao/IShopCardDao.java
package idao;

import entity.Customer;
import entity.ShopCard;

import java.util.LinkedList;
import java.util.List;

public interface IShopCardDao {
    public List<ShopCard> search(Customer customer) throws Exception;
    public int add(Customer customer, ShopCard shopCard) throws Exception;
    public int remove(Customer customer, List<ShopCard> shopCards) throws Exception;
    public int update(Customer customer, ShopCard shopCard)throws Exception;

}

File: src/main/java/idao/IShopCardDao.kt
package idao

import entity.ShopCard

interface IShopCardDao {
    fun insert(item: ShopCard): Int
    fun delete(item: ShopCard): Int
    fun delete(id: Int): Int
    fun update(item: ShopCard): Int
    fun select(name: String): List<ShopCard>
}